lib/package/Bundle.js

```diff
@@ -326,6 +333,69 @@ Bundle.prototype.getReport = function (cb) {
   appendCollection(this.getEndpoints()); // for either apiproxy or sharedflow
   appendCollection(this.getPolicies());
   appendCollection(this.getResources());
+
+  //console.log(JSON.stringify(result, null, 2));
+  const d = DEBUG("apigeelint:directives");
+  if (this.ignoreDirectives) {
+    d(`not considering apigeelint disable directives in files.`);
+  } else {
+    // apply directives to optionally disable some errors/warnings
+    result.forEach((item) => {
+      if (item.messages && item.messages.length) {
+        d(`item ${item.filePath} has ${item.messages.length} messages`);
+        if (
+          item.filePath &&
+          item.filePath.endsWith(".xml") &&
+          isFile(item.filePath)
+        ) {
+          try {
+            const docElement = new Dom().parseFromString(
+              fs.readFileSync(item.filePath).toString()
+            ).documentElement;
+            const directives = myUtil.findDirectives(docElement);
+            d(JSON.stringify(directives, null, 2));
+            if (directives.length > 0) {
+              const keep = (item) => {
+                d("checking " + JSON.stringify(item));
+                const directivesForThisRule = directives.filter((directive) =>
```

Comment:
What if I have `<!-- apigeelint disable=TD004 -->` and the plugin is warning about a different rule say `TD007`?

Will this function raise an issue or just skip it as the directive has the wrong ruleID? I assume it will skip and its the developer's responsibility to use the correct directive. Correct?

Reply:
A disable directive is effective only for the plugin or plugins it specifies. 

If you disable TD004 and TD007 flags a warning, the disable directive has no effect. It's irrelevant. Yes, it is the developer's responsibility to use the correct directive.